### PR TITLE
feat(apigatewayv2): HTTP API - configure CORS preflight

### LINKED ISSUE #7922: Expose CORS Configuration to apigatewayv2.HttpApi
### Use Case

image

#### AS-IS

```
const api = new apigatewayv2.HttpApi(this, 'api')

const _api = api.node.defaultChild as apigatewayv2.CfnApi
_api.corsConfiguration = {
  allowCredentials: true,
  allowHeaders: ...,
  ...
}
```

#### TO-BE

```
new apigatewayv2.HttpApi(this, 'api', {
  corsPreflight: {
    allowCredentials: true,
    allowHeaders: ...,
    ...
  }
}
```

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by civilizeddev**
See #7922

### Commit Message

feat(apigatewayv2): HTTP API - configure CORS preflight (#7923)

- ADD interface CorsPreflightOptions
- ADD HttpApiProps.corsPreflight
- MOD HttpApi.constructor
- MOD README.md

Closes #7922

### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2/test/http/integrations/integ.http-proxy.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ const httpEndpoint = new HttpApi(stack, 'HttpProxyApi', {
   defaultIntegration: new HttpProxyIntegration({
     url: lambdaEndpoint.url!,
   }),
+  corsConfiguration: {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/test/http/integrations/integ.http-proxy.ts:21]: Change to the integ test is not necessary for this change. Revert.

File: packages/@aws-cdk/aws-apigatewayv2/test/http/integrations/http.test.ts [hunk 2]
```diff
@@ -38,4 +38,27 @@ describe('HttpProxyIntegration', () => {
       IntegrationMethod: 'PATCH',
     });
   });
-});
\ No newline at end of file
+
+  test('CORS Configuration is correctly configured.', () => {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/test/http/integrations/http.test.ts:42]: Can you add a test case that `CorsConfiguration` is `ABSENT` when not specified?

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts [hunk 3]
```diff
@@ -36,6 +36,54 @@ export interface HttpApiProps {
...
+
+  /**
+   * Represents a collection of exposed headers.
+   * @default - No Expose Headers are allowed.
+   */
+  readonly exposeHeaders?: string[];
+
+  /**
+   * The number of seconds that the browser should cache preflight request results.
+   * @default 0
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts:84]:
```suggestion
   * The duration that the browser should cache preflight request results.
   * @default Duration.seconds(0)
```

File: packages/@aws-cdk/aws-apigatewayv2/README.md [hunk 4]
```diff
@@ -88,6 +89,31 @@ new HttpApi(stack, 'HttpProxyApi', {
...
+The `corsConfiguration` option lets you specify a CORS configuration for an API.
+
+```ts
+new HttpApi(stack, 'HttpProxyApi', {
+  corsConfiguration: {
+    allowCredentials: true,
+    allowHeaders: ['Authorization'],
+    allowMethods: [HttpMethod.GET, HttpMethod.HEAD, HttpMethod.OPTIONS, HttpMethod.POST],
+    allowOrigins: ['*'],
+    maxAge: 36400,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/README.md:112]: fix

**Comment by nolanrigo** [comment]
Sorry for the extra message, not technically related to this PR.

But I am using CDK since few days and this tool  is a must have.
I was implementing a HttpApi and needed a custom domain + CORS options, I realized it was not implement yet.

What was my surprise when I saw some people already working on it.

I am feeling really grateful for that and I needed to tell you that.

Thank you very much for your work ❤️🙏🚀

File: packages/@aws-cdk/aws-apigatewayv2/README.md [hunk 5]
```diff
@@ -88,6 +89,31 @@ new HttpApi(stack, 'HttpProxyApi', {
...
+The `corsPreflight` option lets you specify a CORS configuration for an API.
+
+```ts
+new HttpApi(stack, 'HttpProxyApi', {
+  corsPreflight: {
+    allowCredentials: true,
+    allowHeaders: ['Authorization'],
+    allowMethods: [HttpMethod.GET, HttpMethod.HEAD, HttpMethod.OPTIONS, HttpMethod.POST],
+    allowOrigins: ['*'],
+    maxAge: Duration.seconds(36400),
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/README.md:112]:
```suggestion
    maxAge: Duration.days(10),
```

**Pull request merged**